### PR TITLE
[Safran-881] Blacklist definition for requirement design

### PR DESCRIPTION
**Pull request by orichalque**
- Extension point definition in requirement.design plugin
- Extension definition in database.design to not display requirements on Sequences semantic decorator.

### CODE REVIEW

File: designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/schema/blacklistedRequirementDecorator.exsd [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      </appInfo>
+      <documentation>
+         # Copyright (c) 2008, 2021 Obeo.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+# 
+# Contributors:
+#     Obeo - initial API and implementation
```
Comment (vrichard12) [inline designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/schema/blacklistedRequirementDecorator.exsd:110]: Please replace with a succinct (yet proper) description of the purpose of this extension point. You can add the copyright as an XML comment at the beginning of the file (you'll find other examples of exsd files in ISD).

File: designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/src/org/obeonetwork/dsl/requirement/design/services/RequirementDecoratorBlacklist.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @author <a href="thibault.beziers-la-fosse@obeo.fr">Thibault Béziers la Fosse</a>
+ *
+ */
+public class RequirementDecoratorBlacklist {
+	
+	private Set<String> blacklistedSemanticDecoratorLabels; 
+	private static final String BLACKLIST_SEMANTIC_DECORATOR_EXTENSION_ID = "org.obeonetwork.dsl.requirement.design.blacklistedRequirementDecorator";
+	private static final String SEMANTIC_DECORATOR_NAME_ATTRIBUTE = "name";
+	
+	public RequirementDecoratorBlacklist() {
```
Comment (vrichard12) [inline designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/src/org/obeonetwork/dsl/requirement/design/services/RequirementDecoratorBlacklist.java:35]: I didn't see where this constructor is called :/

**Review by vrichard12 — COMMENTED** [review]
Please see my comments in the files.

File: designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/src/org/obeonetwork/dsl/requirement/design/services/RequirementDecoratorBlacklist.java (hunk 2 above)
Comment (orichalque) [reply designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/src/org/obeonetwork/dsl/requirement/design/services/RequirementDecoratorBlacklist.java:35]: It is defined as a static attribute in the RequirementServices class, and initialized there

File: designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/schema/blacklistedRequirementDecorator.exsd [hunk 3]
```diff
@@ -0,0 +1,114 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!-- Schema file written by PDE -->
+<schema targetNamespace="org.obeonetwork.dsl.requirement.design" xmlns="http://www.w3.org/2001/XMLSchema">
+<annotation>
+      <appInfo>
+         <meta.schema plugin="org.obeonetwork.dsl.requirement.design" id="blacklistedRequirementDecorator" name="blacklistedRequirementDecorator"/>
+      </appInfo>
+      <documentation>
+         This extension point allows plugins to specify labels of Sirius nodes that should not be decorated with requirements.
+      </documentation>
```
Comment (orichalque) [inline designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/schema/blacklistedRequirementDecorator.exsd:10]: This is the succinct documentatation related to this extension point @vrichard12

File: designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/schema/blacklistedRequirementDecorator.exsd (hunk 1 above)
Comment (orichalque) [reply designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/schema/blacklistedRequirementDecorator.exsd:110]: This is the copyright noticed, I copied & pasted what's in the extension point of /org.obeonetwork.dsl.cinematic

File: designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/src/org/obeonetwork/dsl/requirement/design/services/RequirementDecoratorBlacklist.java (hunk 2 above)
Comment (vrichard12) [reply designs/requirement/plugins/org.obeonetwork.dsl.requirement.design/src/org/obeonetwork/dsl/requirement/design/services/RequirementDecoratorBlacklist.java:35]: ok thanks :)

**Review by vrichard12 — APPROVED** [review]

**Pull request merged**